File: Frontend/angular-app/src/app/modules/admin/repositories/person-phones/commands/create-person-phone-command.ts
import {Inject} from "@angular/core";
import {ServiceResult} from "../../../../../core/models/service-result";
import {IRequest, IRequestHandler} from "../../../../../core/services/mediator/interfaces";
import {HttpService} from "../../../../../core/services/http/http.service";
import {ApplicationError} from "../../../../../core/exceptions/application-error";
import {MediatorHandler} from "../../../../../core/services/mediator/decorator";
import {PersonPhone} from "../../../entities/person-phone";
import {ValidationRule} from "../../../../../core/validation/validation-rule";
import {HttpRequest} from "../../../../../core/services/http/http-request";
import {NotificationService} from "../../../../../shared/services/notification/notification.service";

export class CreatePersonPhoneCommand extends IRequest<CreatePersonPhoneCommand, PersonPhone> {

  public personId: number | undefined = undefined;
  public phoneTypeBaseId: number | undefined = undefined;
  public phoneNumber: string | undefined = undefined;
  public description: string | undefined = undefined;
  public isDefault: boolean | undefined = undefined;

  constructor() {
    super();
  }

  mapFrom(entity: PersonPhone): CreatePersonPhoneCommand {
    throw new ApplicationError(CreatePersonPhoneCommand.name, this.mapTo.name, 'Not Implemented Yet')
  }

  mapTo(): PersonPhone {
    throw new ApplicationError(CreatePersonPhoneCommand.name, this.mapTo.name, 'Not Implemented Yet')
  }

  get url(): string {
    return "/admin/personPhones/add";
  }

  get validationRules(): ValidationRule[] {
    return [];
  }
}

@MediatorHandler(CreatePersonPhoneCommandHandler.name)
export class CreatePersonPhoneCommandHandler implements IRequestHandler<CreatePersonPhoneCommand, PersonPhone> {
  constructor(
    @Inject(HttpService) private _httpService: HttpService,
    @Inject(NotificationService) private _notificationService: NotificationService,

  ) {
  }

  async Handle(request: CreatePersonPhoneCommand): Promise<PersonPhone> {
    let httpRequest: HttpRequest<CreatePersonPhoneCommand> = new HttpRequest<CreatePersonPhoneCommand>(request.url, request);


    return await this._httpService.Post<CreatePersonPhoneCommand, ServiceResult<PersonPhone>>(httpRequest).toPromise().then(response => {
      this._notificationService.showSuccessMessage()
      return response.objResult
    })

  }
}

File: Backend/Microservices/Admin/Eefa.Admin.Application.Refactor/UseCases/PersonPhones/Commands/Create/CreatePersonPhoneCommand.cs
using AutoMapper;
using MediatR;
using System.Threading;
using System.Threading.Tasks;

public class CreatePersonPhoneCommand : IRequest<ServiceResult<PersonPhoneModel>>, IMapFrom<PersonPhone>
{
    public int PersonId { get; set; } = default!;
    public int PhoneTypeBaseId { get; set; } = default!;
    public string PhoneNumber { get; set; }
    public string? Description { get; set; }
    public bool IsDefault { get; set; } = default!;


    public void Mapping(Profile profile)
    {
        profile.CreateMap<CreatePersonPhoneCommand, PersonPhone>();
    }
}

public class CreatePersonPhoneCommandHandler : IRequestHandler<CreatePersonPhoneCommand, ServiceResult<PersonPhoneModel>>
{
    private readonly IUnitOfWork _unitOfWork;
    private readonly IMapper _mapper;

    public CreatePersonPhoneCommandHandler(IUnitOfWork unitOfWork, IMapper mapper)
    {
        _unitOfWork = unitOfWork;
        _mapper = mapper;
    }
    public async Task<ServiceResult<PersonPhoneModel>> Handle(CreatePersonPhoneCommand request, CancellationToken cancellationToken)
    {
        PersonPhone entity = _mapper.Map<PersonPhone>(request);

        _unitOfWork.PersonPhones.Add(entity);
        await _unitOfWork.SaveChangesAsync(cancellationToken);

        return ServiceResult.Success(_mapper.Map<PersonPhoneModel>(entity));
    }
}